exporter/test_exporter.py

```diff
@@ -25,46 +29,141 @@ def client():
+def testExportDatasetTables_InvalidInput(
+    mock_bq_client: mock.MagicMock,
+    mock_split_county: mock.MagicMock,
+    client: FlaskClient
+):
     response = client.post('/', json={})
     assert response.status_code == 400
-
-
-def testExportDatasetTables_NoTables(client: FlaskClient):
-    with patch('google.cloud.bigquery.Client') as mock_bq_client:
-        # Set up mocks
-        mock_bq_instance = mock_bq_client.return_value
-        mock_bq_instance.list_tables.return_value = iter(())
-
-        dataset_name = {'dataset_name': 'my-dataset'}
-        response = client.post('/', json=dataset_name)
-
-        assert response.status_code == 500
-
-
-def testExportDatasetTables_ExtractJobFailure(client: FlaskClient):
-    with patch('google.cloud.bigquery.Client') as mock_bq_client:
-        # Set up mocks
-        mock_bq_instance = mock_bq_client.return_value
-        mock_bq_instance.list_tables.return_value = test_tables
-        mock_extract_job = Mock()
-        mock_bq_instance.extract_table.return_value = mock_extract_job
-        mock_extract_job.result.side_effect = google.cloud.exceptions.InternalServerError('Internal')
-
-        dataset_name = {'dataset_name': 'my-dataset'}
-        response = client.post('/', json=dataset_name)
-
-        assert response.status_code == 500
+    assert mock_split_county.call_count == 0
+
+
+@mock.patch('main.export_split_county_tables')
+@mock.patch('google.cloud.bigquery.Client')
+def testExportDatasetTables_NoTables(
+    mock_bq_client: mock.MagicMock,
+    mock_split_county: mock.MagicMock,
+    client: FlaskClient
+):
+    # Set up mocks
+    mock_bq_instance = mock_bq_client.return_value
+    mock_bq_instance.list_tables.return_value = iter(())
+
+    dataset_name = {'dataset_name': 'my-dataset'}
+    response = client.post('/', json=dataset_name)
+
+    assert response.status_code == 500
+    assert mock_split_county.call_count == 0
+
+
+@mock.patch('main.export_split_county_tables')
+@mock.patch('google.cloud.bigquery.Client')
+def testExportDatasetTables_ExtractJobFailure(
+    mock_bq_client: mock.MagicMock,
+    mock_split_county: mock.MagicMock,
+    client: FlaskClient
+):
+    # Set up mocks
+    mock_bq_instance = mock_bq_client.return_value
+    mock_bq_instance.list_tables.return_value = test_tables
+    mock_extract_job = Mock()
+    mock_bq_instance.extract_table.return_value = mock_extract_job
+    mock_extract_job.result.side_effect = google.cloud.exceptions.InternalServerError(
+        'Internal')
+
+    dataset_name = {'dataset_name': 'my-dataset'}
+    response = client.post('/', json=dataset_name)
+
+    assert response.status_code == 500
+    assert mock_split_county.call_count == 1
+
+#
+#
+#
+
+
+# TEST ADDITIONAL COUNTY-LEVEL DATASET SPLIT FUNCTIONS
+_test_query_results_df = pd.DataFrame({
+    'county_fips': ["01001", "01002", "01003"],
+    'some_condition_per_100k': [None, 1, 2],
+})
+
+
+def _get_query_results_as_df(*args):
```

Comment:
This is the right idea but I think

* All of the assertions should happen inside the test function. You should take the assertions out of the side effect function, and instead inspect the args list to make sure that the query is correct.
* You should make sure that the query string is exactly what you would expect. 

Looks great though.